nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/FetchHDFS.java

```diff
@@ -116,10 +123,40 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             return;
         }
 
+        InputStream stream = null;
+        CompressionCodec codec = null;
+        Configuration conf = getConfiguration();
+        final CompressionCodecFactory compressionCodecFactory = new CompressionCodecFactory(conf);
+        final CompressionType compressionType = CompressionType.valueOf(context.getProperty(COMPRESSION_CODEC).toString());
+        final boolean inferCompressionCodec = compressionType == CompressionType.AUTOMATIC;
+
+        if (inferCompressionCodec || compressionType != CompressionType.NONE) {
+            codec = getCompressionCodec(context, getConfiguration());
+        }
+
         final URI uri = path.toUri();
         final StopWatch stopWatch = new StopWatch(true);
-        try (final FSDataInputStream inStream = hdfs.open(path, 16384)) {
-            flowFile = session.importFrom(inStream, flowFile);
+        try {
+
+            final String outputFilename;
+            final String originalFilename = path.getName();
+            stream = hdfs.open(path, 16384);
+
+            // Check if we should infer compression codec
+            if (inferCompressionCodec) {
+                codec = compressionCodecFactory.getCodec(path);
```

Comment:
I am not sure I understand what;s happening here in relation to line:134. Since both IF statements evaluate ```inferCompressionCodec``` this will always override the value of 'codec' (the one that was set in line:134) if it's ```true```.